bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/obsComp/PwfsGuideProbe.java

```diff
@@ -284,41 +283,48 @@ public GuideStarValidation validate(Coordinates coords, ObsContext ctx) {
-     * inside, if inside the inner limit and innerBoundary if between the inner and outer limits
-     * (which is where the PWFS should normally be).
+     * Returns:
+     *
+     * - BoundaryCheck.outOfRange if outside the limit.
+     *
+     * - BoundaryCheck.vignetting if it obscures the science area.
+     *
+     * - BoundaryCheck.inRange if inside the limit.
      *
      * @param coords the coordinates
      * @param ctx the context
      */
-    public Option<BoundaryPosition> checkBoundaries(final Coordinates coords, final ObsContext ctx) {
+    public Option<PwfsProbeRangeArea> checkBoundaries(final Coordinates coords, final ObsContext ctx) {
         return ctx.getBaseCoordinates().map(baseCoordinates -> {
             final Angle positionAngle = ctx.getPositionAngleJava();
             final Set<Offset> sciencePositions = ctx.getSciencePositions();
 
             // check positions against corrected outer patrol field bounds
             return getCorrectedPatrolField(ctx).map(pf -> {
                 final BoundaryPosition bp = pf.checkBoundaries(coords, baseCoordinates, positionAngle, sciencePositions);
-                if (bp != BoundaryPosition.inside) {
-                    return BoundaryPosition.outside;
-                }
-
-                // Check if any of the guide stars are inside the inner bounds (opposite logic needed, union instead of intersection)
-                final double minLimit = getVignettingClearance(ctx).toArcsecs().getMagnitude();
-                final Ellipse2D e = new Ellipse2D.Double(-minLimit, -minLimit, minLimit * 2.0, minLimit * 2.0);
-                final PatrolField p = getCorrectedPatrolField(new PatrolField(e, e, e), ctx);
-                if (p.anyInside(coords, baseCoordinates, positionAngle, sciencePositions)) {
-                    return BoundaryPosition.inside;
+                switch (bp) {
+                    case outside:
+                    case outerBoundary:
+                        return PwfsProbeRangeArea.outOfRange;
+                    case inside:
+                    case innerBoundary:
+                        // Check if any of the guide stars are inside the inner bounds (opposite logic needed, union instead of intersection)
+                        final double minLimit = getVignettingClearance(ctx).toArcsecs().getMagnitude();
+                        final Ellipse2D e = new Ellipse2D.Double(-minLimit, -minLimit, minLimit * 2.0, minLimit * 2.0);
+                        final PatrolField p = getCorrectedPatrolField(new PatrolField(e, e, e), ctx);
+                        if (p.anyInside(coords, baseCoordinates, positionAngle, sciencePositions)) {
+                            return PwfsProbeRangeArea.vignetting;
+                        }
                 }
-                return BoundaryPosition.innerBoundary;
-            }).getOrElse(BoundaryPosition.outside);
+                return PwfsProbeRangeArea.inRange;
```

Comment:
I tried to include this return statement in the `inside`/`innerBoundary` switch case but it didn't work.

Reply:
Perhaps the way to do it in Java is to declare a `final PwfsProbeRangeArea result;` before the switch statement, assign it in all cases inside the switch statement, and just have the one return outside.